### PR TITLE
fix: CogVideox train dataset _preprocess_data crop video

### PR DESCRIPTION
**Pull request by glide-the**
-  Removed int8 to float32 conversion (`* 2.0 - 1.0`) from `train_transforms` as it caused image overexposure.

-  Added `_resize_for_rectangle_crop` function to enable video cropping functionality. The cropping mode can be configured via `video_reshape_mode`, supporting options: ['center', 'random', 'none'].

image

@sayakpaul

### CODE REVIEW

**Comment by sayakpaul** [comment]
Cc: @a-r-r-o-w

**Review by a-r-r-o-w — COMMENTED** [review]
Thanks, I have one question. Apart from that, this looks great!

cc @yiyixuxu if we want the ipynb notebook here or not

File: examples/cogvideo/train_cogvideox_lora.py [hunk 1]
```diff
@@ -576,10 +620,12 @@ def _preprocess_data(self):
             assert (selected_num_frames - 1) % 4 == 0
 
             # Training transforms
-            frames = frames.float()
-            frames = torch.stack([train_transforms(frame) for frame in frames], dim=0)
-            videos.append(frames.permute(0, 3, 1, 2).contiguous())  # [F, C, H, W]
+            tensor = frames.float() / 255.0
```
[the remark this answers is not in the document]
Comment (glide-the) [reply examples/cogvideo/train_cogvideox_lora.py:623]: > I am not sure why we are making the tensors to range `[0, 1]`, instead of `[-1, 1]`. In the original codebase, we convert to `[-1, 1]` as well [here](https://github.com/THUDM/CogVideo/blob/111756a6a68a8df375ef9c31f9f325818699dfaa/sat/data_video.py#L437) if I understand correctly, yes?

You're right, it should be in the [-1, 1] range. In fact, this is for matrix calculations during fine-tuning, and the [-1, 1] range is easier for computation. I forgot that this step is handled in the latten2img process, so the image is in the [0, 1] range, while the latent space is in the [-1, 1] range.

I've already verified that the cause of the training result showing a blank screen is that I input a 960x720 image into the dataset, and it was compressed to a 460x720 image for training directly.

[the remark this answers is not in the document]
Comment (glide-the) [reply examples/cogvideo/train_cogvideox_lora.py:623]: https://github.com/THUDM/CogVideo/blob/111756a6a68a8df375ef9c31f9f325818699dfaa/sat/data_video.py#L437 
The number 127.5 may experience precision loss during division operations.

**Review by a-r-r-o-w — COMMENTED** [review]
Thanks for the improvements! Have you verified training on the new settings? I think it would be good to default `--video_reshape_mode` to maintain compatibility with existing `*.sh` scripts for launching training that others might have locally setup.

Could you host the ipynb notebook on https://gist.github.com/ instead of here, and link to it instead? We try to limit notebooks, pngs/mp4s/gifs and other files in the repo otherwise it can soon compound to a bulky clone.

I think `make style` is also needed to quality tests

**Comment by glide-the** [comment]
> Thanks for the improvements! Have you verified training on the new settings? I think it would be good to default `--video_reshape_mode` to maintain compatibility with existing `*.sh` scripts for launching training that others might have locally setup.
> 
> Could you host the ipynb notebook on https://gist.github.com/ instead of here, and link to it instead? We try to limit notebooks, pngs/mp4s/gifs and other files in the repo otherwise it can soon compound to a bulky clone.
> 
> I think `make style` is also needed to quality tests
move video_fix_rgb_float_and_crop.ipynb to  https://gist.github.com/glide-the/7658dbfd5f555be0a1a687a4139dba40

File: examples/cogvideo/README.md [hunk 2]
```diff
@@ -180,6 +180,7 @@ Note that setting the `<ID_TOKEN>` is not necessary. From some limited experimen
 
 > [!TIP]
 > You can pass `--use_8bit_adam` to reduce the memory requirements of training.
+> You can pass `--video_reshape_mode` video cropping functionality, supporting options: ['center', 'random', 'none'].
```
Comment (a-r-r-o-w) [inline examples/cogvideo/README.md:183]:
```suggestion
> You can pass `--video_reshape_mode` video cropping functionality, supporting options: ['center', 'random', 'none']. See [this](https://gist.github.com/glide-the/7658dbfd5f555be0a1a687a4139dba40) notebook for examples.
```

**Review by a-r-r-o-w — APPROVED** [review]
